### DISCUSSION TITLE
Concerns about the recent "Split UCI into UCIEngine and Engine" patch

### DISCUSSION THREAD

**Discussion by Coolchessguykevin** [official-stockfish/Stockfish#5170] [Q&A] [answered]
Dear Stockfish developers and experts,

I have some concerns about the recent patch ("Split UCI into UCIEngine and Engine").

Please tell me, will Stockfish now work correctly in all the GUIs in which it worked previously? Including the Aquarium 2020 GUI. Will the Aquarium accept the Stockfish engine correctly with these innovations? In particular, correctly display output PVs, correctly perceive changes in the Stockfish UCI parameters through the GUI.

I see that there have already been several serious (judging by the description) bugfixes (such as “Fix wrong mate sign, Fix some more UCI output and Remove duplicated Position object in UCIEngine”), and I'm worried about how safe it is to use the current Stockfish in GUIs, especially these with old versions. 

In general, will everything work properly, as it worked before, in all chess GUIs? How fundamental are the changes in the interpretation of Stockfish UCI for chess graphical shells? Will the analysis and output PV still be correct and all UCI parameters work correctly? Or should I wait for new patches?

I apologize if my concerns are unfounded. Thank you.

Best regards, Ivan.

**Comment by Disservin** [comment, marked as answer]
There is no (should) real change in what the GUI's see, for them stockfish is the same stockfish it was before.

**Reply by Disservin** [reply]
A few bugs got in unfortunately but those were resolved rather quick afterwards, so all is normal.

**Reply by Coolchessguykevin** [reply]
Dear Disservin,
 
Thank you very much for the clarification! Now I'm calm.
 
I even ran a small 1+0 time control match in Aquarium GUI (the pre "Split UCI into UCIEngine and Engine" patch version («Update NNUE architecture to SFNNv9 and net nn-ae6a388e4a1a.nnue») and the current dev version ("Fix wrong sign for 200 TB score"), both engines in the match have the same parameters that I usually use in regular position analyzes in the Aquarium GUI) to eliminate obvious bugs that could potentially affect the strength of the play. After 200 games the result is good (see screenshot).
 
Later I will also study the output PVs in analysis mode and let you know if I find any problems in any newest dev versions.
 
Thank you and all the other developers for your continued diligence and fruitful development of the Stockfish chess engine!
 
Best regards,
Ivan. 
![SF test UCI vs refactor UCI](https://github.com/official-stockfish/Stockfish/assets/44373850/043bba04-2c31-48c3-8584-baaa72dff9ca)
